src/middleware/router.js

```diff
@@ -450,16 +453,69 @@ function sendHttpRequest (ctx, route, options) {
           uncompressedBodyBufs.push(data)
         })
       }
+*/
+//      const bufs = []
+ 
+      if (isNaN(counter)) {
+        counter = 0
+        size = 0
+
+        if(!bucket) {
+          bucket = getGridFSBucket()
+        }
 
-      const bufs = []
+        ctx.response.body = new Writable({
+          write: function(chunk, encoding, next) {
+            counter++
+            size += chunk.toString().length
+            console.log(`Write Response CHUNK # ${counter} upstream [ Cum size ${size} ]`)
+            next()
+          }
+        }).on('error', (err) => {
+            logger.error('Error sending Response upstream: ' + JSON.stringify(err))
+            reject(err)
+          })
+          .on('finish', () => {
+            logger.info(`Finished sending Response upstream`)
+            resolve(response)
+          })
+
+        uploadStream = bucket.openUploadStream()
```

Comment:
Could we not maybe re-use the contentChunk function already exists, instead of writing all of this again. If it functions differently then we might need to re-think /refactor the contentChunk functions